src/pyscaffold/extensions/interactive.py

```diff
@@ -0,0 +1,282 @@
+"""Similarly to ``git rebase -i`` this extension allows users to interactively
+choose which options apply to ``putup``, by editing a file filled with examples.
+
+See :obj:`CONFIG` for more details on how to tweak the text generated in the interactive
+mode.
+
+.. versionadded:: 4.0
+   *"interactive mode"* introduced as an **experimental** extension.
+
+Warning:
+
+    **NOTE FOR CONTRIBUTORS**:
+    Due to the way :mod:`argparse` is written, it is not very easy to obtain information
+    about which options and arguments a given parser is currently configured with.
+    There are no public methods that allow inspection/reflection, and therefore in order
+    to do so, one has to rely in a few non-public methods (according to Python's
+    convention, the ones starting with a ``_`` symbol).
+    Since :mod:`argparse` implementation is very stable and mature, these non-public
+    method are very unlikely to change and, therefore, it is relatively safe to use
+    these methods, however developers and maintainers have to be aware and pay attention
+    to eventual breaking changes.
+    The non-public functions are encapsulated in the functions :obj:`get_actions` and
+    :obj:`format_args` in this file, in order to centralise the usage of non-public API.
+"""
+import os
+import shlex
+import textwrap
+from argparse import Action, ArgumentParser
+from collections import abc
+from functools import lru_cache, reduce
+from itertools import chain
+from typing import List, Optional, Set
+
+from .. import api, cli, file_system, shell, templates
+from ..actions import ScaffoldOpts as Opts
+from ..actions import get_default_options
+from . import Extension
+from . import list_from_entry_points as list_all_extensions
+
+INDENT_LEVEL = 4
+HEADER = templates.get_template("header_interactive")
+
+
+CONFIG_KEY = "interactive"
+CONFIG = {
+    "ignore": ["--help", "--version"],
+    "comment": ["--verbose", "--very-verbose"],
+}
+"""Configuration for the options that are not associated with an extension class.
+This dict value consist of a set of metadata organised as follows:
+
+- Each value must be a list of "long" :obj:`argparse` option strings (e.g. `"--help"`
+  instead of `"-h"`).
+- Each key implies on a different interpretation for the metadata:
+
+    - ``"ignore"``: Options that should be simply ignored when creating examples
+    - ``"comment"``: Options that should be commented when creating examples,
+      even if they appear in the original :obj:`sys.argv`.
+
+Extension classes (or instances) can also provide configurations by defining a
+``interactive`` attribute assigned to a similar :obj:`dict` object.
+"""
+
+
+@lru_cache(maxsize=2)
```

Comment:
Does this speed things up a lot? Just asking as introducing caching can always lead to quite subtle bugs later on.

Reply:
So the idea is that I call this function for every action, and everytime it loads the extensions from the entrypoints (so Python goes to setuptools files collecting entries and then do some just-in-time file loading). I am not sure how long it takes, I added the cache there just intuitively to be sincere. I agree the best would be to measure before doing any premature optimisation 😅 

To be fair, once all the extension classes and entrypoints are defined, this thing should be pretty much immutable (it is config metadata after all and works as a "distributed" constant across the extensions)... There could be problems if we are messing around with stubbing and mocking in the tests, but instead I am stubbing the `get_config` function itself instead of the things it depends on, so the cache is also taken away.